pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -354,19 +360,101 @@ func (r *ReconcileInfinispan) deploymentForInfinispan(m *infinispanv1.Infinispan
 			},
 		},
 	}
-
+	setupVolumesForEncryption(m, dep)
 	//	appendVolumes(m, dep)
 
 	// Set Infinispan instance as the owner and controller
 	controllerutil.SetControllerReference(m, dep, r.scheme)
 	return dep
 }
 
+func getEncryptionSecretName(m *infinispanv1.Infinispan) string {
+	ee := m.Spec.Security.EndpointEncryption
+	if ee.Type == "service" {
+		if strings.Contains(ee.CertService, "openshift.io") {
+			// Using platform service. Only OpenShift is integrated atm
+			return "service-certs"
+		}
+	}
+	return ee.CertSecretName
+}
+
+func setupServiceForEncryption(m *infinispanv1.Infinispan, ser *corev1.Service) {
+	ee := m.Spec.Security.EndpointEncryption
+	if ee.Type == "service" {
+		if strings.Contains(ee.CertService, "openshift.io") {
+			// Using platform service. Only OpenShift is integrated atm
+			secretName := getEncryptionSecretName(m)
+			if ser.ObjectMeta.Annotations == nil {
+				ser.ObjectMeta.Annotations = map[string]string{}
+			}
+			ser.ObjectMeta.Annotations[ee.CertService] = secretName
+		}
+	}
+}
+
+func setupVolumesForEncryption(m *infinispanv1.Infinispan, dep *appsv1beta1.StatefulSet) error {
```

Comment:
Why does this function return error? It always returns nil. Unless a non-nil error can be thrown, I'd make it a void function.